cli/cmd/proxy/list/command.go

```diff
@@ -0,0 +1,218 @@
+package list
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/consul-k8s/cli/common"
+	"github.com/hashicorp/consul-k8s/cli/common/flag"
+	"github.com/hashicorp/consul-k8s/cli/common/terminal"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ListCommand is the command struct for the proxy list command.
+type ListCommand struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagNamespace     string
+	flagAllNamespaces bool
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+// init sets up flags and help text for the command.
+func (c *ListCommand) init() {
+	c.set = flag.NewSets()
+
+	f := c.set.NewSet("Command Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "namespace",
+		Target:  &c.flagNamespace,
+		Default: "default",
```

Comment:
I think the idea that the Default namespace is `default` makes sense on the surface, but shouldn't it actually be the default namespace defined by the user's kubecontext?

Reply:
Yep! That was a good call. I made that adjustment and tested that it uses the namespace set by kubeconfig.